File: articles/warren-mcculloch-walter-pitts-and-neural-logic/c/threshold_unit.c
#include <stdio.h>

int threshold_unit(const int inputs[], const int weights[], int n, int threshold) {
    int total = 0;
    for (int i = 0; i < n; ++i) total += inputs[i] * weights[i];
    return total >= threshold ? 1 : 0;
}

int main(void) {
    int inputs[] = {1, 1};
    int weights[] = {1, 1};
    printf("%d\n", threshold_unit(inputs, weights, 2, 2));
    return 0;
}

File: articles/warren-mcculloch-walter-pitts-and-neural-logic/go/threshold_unit.go
package main

import "fmt"

func thresholdUnit(inputs []int, weights []int, threshold int) int {
	total := 0
	for i := range inputs {
		total += inputs[i] * weights[i]
	}
	if total >= threshold {
		return 1
	}
	return 0
}

func main() {
	fmt.Println(thresholdUnit([]int{1, 1}, []int{1, 1}, 2))
}
